src/snowflake/snowpark/_internal/analyzer/select_statement.py

```diff
@@ -383,6 +383,44 @@ def referenced_ctes(self) -> Dict[WithQueryBlock, int]:
         reference count of the cte. Includes itself and its children"""
         pass
 
+    def merge_into_pre_action(self, pre_action: "Query") -> None:
+        """Method to merge a pre-action into the current Selectable's pre-actions if it
+        is not already present. If pre_actions is None, new list will be initialized."""
+        if self.pre_actions is None:
+            self.pre_actions = [copy(pre_action)]
+        elif pre_action not in self.pre_actions:
+            self.pre_actions.append(copy(pre_action))
+
+    def merge_into_post_action(self, post_action: "Query") -> None:
+        """Method to merge a post-action into the current Selectable's post-actions if it
+        is not already present. If post_actions is None, new list will be initialized."""
+        if self.post_actions is None:
+            self.post_actions = [copy(post_action)]
+        elif post_action not in self.post_actions:
+            self.post_actions.append(copy(post_action))
+
+    def with_subqueries(
+        self, subquery_plans: List[SnowflakePlan], snowflake_plan: SnowflakePlan
+    ) -> "Selectable":
+        """Update pre-actions, post-actions and schema to capture necessary subquery_plans
+        encountered during plan resolution. All updates are in-place.
+
+        Args:
+            subquery_plans: List of subquery plans encountered during plan resolution.
+            snowflake_plan: The snowflake plan which is created and updated using
```

Comment:
-> the snowflake corresponding to the fully resolved snowflake plan for the current selectable with the subquery plan handled